code/game/objects/structures/inflatable.dm

```diff
@@ -0,0 +1,259 @@
+		deflate(1)
+	return
+
+/obj/structure/inflatable/ex_act(severity)
+	switch(severity)
+		if(1.0)
+			qdel(src)
+			return
+		if(2.0)
+			deflate(1)
+			return
+		if(3.0)
+			if(prob(50))
+				deflate(1)
+				return
+
+/obj/structure/inflatable/blob_act()
+	deflate(1)
+
+/obj/structure/inflatable/attack_hand(mob/user as mob)
+	add_fingerprint(user)
+	return
+
+/obj/structure/inflatable/attackby(obj/item/weapon/W as obj, mob/user as mob, params)
+	if(!istype(W))
+		return
+	if(W.is_sharp() || is_pointed(W))
+		visible_message("<span class='danger'>[user] pierces [src] with [W]!</span>")
+		deflate(1)
+	if(W.damtype == BRUTE || W.damtype == BURN)
+		hit(W.force)
+		..()
+	return
+
+/obj/structure/inflatable/proc/hit(var/damage, var/sound_effect = 1)
+	health = max(0, health - damage)
+	if(sound_effect)
+		playsound(loc, 'sound/effects/Glasshit.ogg', 75, 1)
+	if(health <= 0)
+		deflate(1)
+
+/obj/structure/inflatable/AltClick()
+	if(usr.stat || usr.restrained())
+		return
+	if(!Adjacent(usr))
+		return
+	deflate()
+
+/obj/structure/inflatable/proc/deflate(var/violent=0)
+	playsound(loc, 'sound/machines/hiss.ogg', 75, 1)
+	if(violent)
+		visible_message("[src] rapidly deflates!")
+		var/obj/item/inflatable/torn/R = new /obj/item/inflatable/torn(loc)
+		src.transfer_fingerprints_to(R)
+		qdel(src)
+	else
+		visible_message("[src] slowly deflates.")
+		spawn(50)
+			var/obj/item/inflatable/R = new /obj/item/inflatable(loc)
+			src.transfer_fingerprints_to(R)
+			qdel(src)
+
+/obj/structure/inflatable/verb/hand_deflate()
+	set name = "Deflate"
+	set category = "Object"
+	set src in oview(1)
+
+	if(usr.stat || usr.restrained())
+		return
+
+	deflate()
+
+/obj/item/inflatable/door
+	name = "inflatable door"
+	desc = "A folded membrane which rapidly expands into a simple door on activation."
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "folded_door"
+
+/obj/item/inflatable/door/attack_self(mob/user)
+	playsound(loc, 'sound/items/zip.ogg', 75, 1)
+	to_chat(user, "<span class='notice'>You inflate [src].</span>")
+	var/obj/structure/inflatable/door/R = new /obj/structure/inflatable/door(user.loc)
+	src.transfer_fingerprints_to(R)
+	R.add_fingerprint(user)
+	qdel(src)
+
+/obj/structure/inflatable/door //Based on mineral door code
+	name = "inflatable door"
+	density = 1
+	anchored = 1
+	opacity = 0
+
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "door_closed"
+
+	var/state = 0 //closed, 1 == open
+	var/isSwitchingStates = 0
+
+/obj/structure/inflatable/door/attack_ai(mob/user as mob) //those aren't machinery, they're just big fucking slabs of a mineral
+	if(isAI(user)) //so the AI can't open it
+		return
+	else if(isrobot(user)) //but cyborgs can
+		if(get_dist(user,src) <= 1) //not remotely though
+			return TryToSwitchState(user)
+
+/obj/structure/inflatable/door/attack_hand(mob/user as mob)
+	return TryToSwitchState(user)
+
+/obj/structure/inflatable/door/CanPass(atom/movable/mover, turf/target, height=0)
+	if(istype(mover, /obj/effect/beam))
```

Comment:
Probably? I changed the return from "return !opacity" to "return 1".

Was that the issue or is the if(istype) check just totally unnecessary?